### PR TITLE
[Storage] Support DatalakeGen2 GA, and upgrade Blob to Track2 SDK

### PR DESCRIPTION
**Pull request by blueww**
## Description

Design:  https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/550 
Still need Fix from Erich, so don't merge.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: tools/StaticAnalysis/Exceptions/Az.Storage/MissingAssemblies.csv [hunk 1]
```diff
@@ -2,4 +2,14 @@
...
+"Az.Storage","System.Numerics.Vectors","4.1.3.0","System.Text.Json","0","3000","Missing assembly System.Numerics.Vectors referenced from System.Text.Json","Ensure that the assembly is included in the Wix file or directory"
+"Az.Storage","System.Buffers","4.0.2.0","System.Text.Json","0","3000","Missing assembly System.Buffers referenced from System.Text.Json","Ensure that the assembly is included in the Wix file or directory"
+"Az.Storage","System.Runtime.CompilerServices.Unsafe","4.0.5.0","System.Text.Json","0","3000","Missing assembly System.Runtime.CompilerServices.Unsafe referenced from System.Text.Json","Ensure that the assembly is included in the Wix file or directory"
+"Az.Storage","System.Text.Encodings.Web","4.0.4.0","System.Text.Json","0","3000","Missing assembly System.Text.Encodings.Web referenced from System.Text.Json","Ensure that the assembly is included in the Wix file or directory"
+"Az.Storage","System.Runtime.CompilerServices.Unsafe","4.0.5.0","System.Text.Json","0","3000","Missing assembly System.Runtime.CompilerServices.Unsafe referenced from System.Text.Json","Ensure that the assembly is included in the Wix file or directory"
```
Comment (erich-wang) [inline tools/StaticAnalysis/Exceptions/Az.Storage/MissingAssemblies.csv:15]: Ideally we should not add these items here, I'm double checking locally.

Comment (blueww) [reply tools/StaticAnalysis/Exceptions/Az.Storage/MissingAssemblies.csv:15]: These are added by Mark in https://github.com/wastoresh/azure-powershell/pull/5

Comment (erich-wang) [reply tools/StaticAnalysis/Exceptions/Az.Storage/MissingAssemblies.csv:15]: Please change:
```suggestion
 "Az.Storage","System.Text.Encodings.Web","4.0.4.0","System.Text.Json","0","3000","Missing assembly System.Text.Encodings.Web referenced from System.Text.Json","Ensure that the assembly is included in the Wix file or directory"
  "Az.Storage","System.Numerics.Vectors","4.1.3.0","System.Text.Json","0","3000","Missing assembly System.Numerics.Vectors referenced from System.Text.Json","Ensure that the assembly is included in the Wix file or directory"
  "Az.Storage","System.Runtime.CompilerServices.Unsafe","4.0.5.0","System.Text.Json","0","3000","Missing assembly System.Runtime.CompilerServices.Unsafe referenced from System.Text.Json","Ensure that the assembly is included in the Wix file or directory"
```

**Pull request merged**